src/components/Button/ButtonAddPostTopic.js

```diff
@@ -1,29 +1,32 @@
 import * as React from 'react';
 import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
 
+import PropTypes from 'prop-types';
 import PostToCommunity from '../../assets/icon/PostToCommunity';
 import useBetterNavigationHook from '../../hooks/navigation/useBetterNavigationHook';
 import dimen from '../../utils/dimen';
 import {normalize} from '../../utils/fonts';
 import ShadowFloatingButtons from './ShadowFloatingButtons';
 import {COLORS} from '../../utils/theme';
 
-const ButtonAddPostTopic = ({topicName, onRefresh}) => {
+const ButtonAddPostTopic = ({topicName, onRefresh, followType}) => {
   const {toCreatePostWithTopic} = useBetterNavigationHook();
 
   const onAddPostPressed = () => {
-    toCreatePostWithTopic(topicName, {onRefresh});
+    toCreatePostWithTopic(topicName, {onRefresh, followType});
   };
 
   return (
     <View style={styles.container}>
       <ShadowFloatingButtons>
         <TouchableOpacity
           onPress={onAddPostPressed}
-          style={styles.buttonContainer}
+          style={styles.buttonContainer(followType)}
           testID="onaddtopicbutton">
           <View style={styles.postToCommunityContainer}>
-            <PostToCommunity />
+            <PostToCommunity
+              color={followType === 'incognito' ? COLORS.anon_secondary : COLORS.signed_primary}
+            />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ButtonAddPostTopic` component now accepts a new prop `followType`. This is used to conditionally style the button and set the color of the `PostToCommunity` icon. The logic seems correct, but it's important to ensure that the `followType` prop is always provided where this component is used, or else handle its absence gracefully.
```diff
- const ButtonAddPostTopic = ({topicName, onRefresh}) => {
+ const ButtonAddPostTopic = ({topicName, onRefresh, followType}) => {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->